Lib/test/test__xxsubinterpreters.py

```diff
@@ -526,30 +526,25 @@ def test_with_int(self):
         self.assertEqual(int(id), 10)
 
     def test_coerce_id(self):
-        id = interpreters.InterpreterID('10', force=True)
-        self.assertEqual(int(id), 10)
-
-        id = interpreters.InterpreterID(10.0, force=True)
-        self.assertEqual(int(id), 10)
-
         class Int(str):
-            def __init__(self, value):
-                self._value = value
-            def __int__(self):
-                return self._value
+            def __index__(self):
```

Comment:
`__index__` makes sense, but what about `__int__` too.  Shouldn't both work?

Reply:
`float` has `__int__`, but not `__index__`. `(10.1).__int__()` returns `10`. So `__int__` should not work.

Note that I changed `return self._value` to `return 10` because `int(10)` and `int(str(10))` has the same result. So it was not clear what way the `str` subclass with `__int__` used, bot has the same result.